server/src/main/java/org/apache/druid/segment/loading/SegmentLoaderConfig.java

```diff
@@ -88,19 +92,32 @@ public int getNumBootstrapThreads()
     return numBootstrapThreads == null ? numLoadingThreads : numBootstrapThreads;
   }
 
+  public StorageLocationSelectorStrategy getStorageLocationSelectorStrategy(List<StorageLocation> storageLocations)
+  {
+    if (locationSelectorStrategy == null) {
+      checkLocationConfigForNull();
+      // default strategy if no strategy is specified in the config
+      locationSelectorStrategy = new LeastBytesUsedStorageLocationSelectorStrategy(storageLocations);
+    }
+    return locationSelectorStrategy;
+  }
+
   public File getInfoDir()
   {
     if (infoDir == null) {
-
-      if (locations == null || locations.size() == 0) {
-        throw new ISE("You have no segment cache locations defined. Please configure druid.segmentCache.locations to use one or more locations.");
-      }
+      checkLocationConfigForNull();
       infoDir = new File(locations.get(0).getPath(), "info_dir");
     }
-
     return infoDir;
   }
 
+  private void checkLocationConfigForNull()
```

Comment:
can you check if putting `@NotNull` on `locations` automatically does that ?

Reply:
locations already has @NotEmpty. This alone should work. But an IllegalStateException's being thrown when locations is null or empty with an error message. As an IllegalStateException is being thrown explicitly didn't want to change this.

Reply:
Removed checkLocationConfigForNull() method as `@NotEmpty` annotation for `locations` is serving the purpose.